deployment/src/main/java/io/quarkus/jsch/deployment/JSchProcessor.java

```diff
@@ -131,4 +154,65 @@ ReflectiveClassBuildItem reflection() {
                 "com.jcraft.jsch.UserAuthPublicKey")
                 .fields().methods().build();
     }
+
+    @BuildStep
+    void registerAdditionalBeans(BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
+        additionalBeans.produce(AdditionalBeanBuildItem.builder()
+                .addBeanClass(JSchSessions.class)
+                .setUnremovable()
+                .setDefaultScope(DotName.createSimple(RequestScoped.class))
```

Comment:
Making it `@RequestScoped` will recreate the `JSchSessions` on every request, is that intentional?

Reply:
I have hesitated between between 2 solutions:

* use a `RequestScoped` bean to close as soon as possible all not needed connections
* use an `ApplicationScoped` bean and add a *keepAlive* mechanism

Reply:
Here is one (crazy?) idea: while you're iterating through the fields annotated with `@JSchSession` (as you do in `produceSessions`), produce an injection point with a `@Dependent` scope: so the classes that inject them can decide which scope to use. WDYT?

Reply:
good idea, but I have to introduce a keep alive mechanism if the injector class is an ApplicationScoped or a Singleton

Reply:
Would it work if we made it a synthetic bean that calls the `Session#disconnect()` when the scope it is injected is destroyed?

Reply:
Now, it is `Dependant` and there is a keep alive which is disabled by default